spring-integration-ftp/src/main/java/org/springframework/integration/ftp/outbound/FtpMessageHandler.java

```diff
@@ -60,7 +64,7 @@ public boolean isChmodCapable() {
 	@Override
 	protected void doChmod(RemoteFileTemplate<FTPFile> remoteFileTemplate, final String path, final int chmod) {
 		remoteFileTemplate.executeWithClient((ClientCallbackWithoutResult<FTPClient>) client -> {
-			String chModCommand = "chmod " + Integer.toOctalString(chmod) + " " + path;
+			String chModCommand = "chmod " + chmod + " " + path;
```

Comment:
What made you to do this radical architectural change?

I think that is really a point to perform `chmod` command with *octal*.

Reply:
Already reverted!